### PR TITLE
DDF-3633 updated docs for metacard attribute editing

### PR DESCRIPTION
**Pull request by ricklarsen**
**PORT to `master` of PR https://github.com/codice/ddf/pull/3025**

#### What does this PR do?

Updates documentation in these areas:
- Configuring: showing hiding metacards with data validation errors from users.
- Data management: editing and overriding metacards
- Developing: edits to creating an attribute injection file.

#### Who is reviewing it? 

@brendan-hofmann @ahoffer @mcalcote @vinamartin 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams/data/members

#### Choose 2 committers to review/merge the PR. 
@beyelerb
@brendan-hofmann
@clockard
@coyotesqrl
@lessarderic
@ricklarsen - Documentation
@shaundmorris

#### How should this be tested? (List steps with links to updated documentation)

review content additions and edits.

#### What are the relevant tickets?
[DDF-3633](https://codice.atlassian.net/browse/DDF-3633)

#### Checklist:
- [x] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
build now

**Comment by ricklarsen** [comment]
build now

File: distribution/docs/src/main/resources/content/_devComponents/attribute-injection.adoc [hunk 1]
```diff
@@ -40,23 +42,68 @@ In that case it can be thought of as a 'global' attribute injection.
...
+      "attribute": "cloud-cover",
+      "metacardTypes": "video"
+    }
+  ]
 }
 ----
 
 [NOTE]
 ====
 Attributes must be registered in the attribute registry (see the `AttributeRegistry` interface) in order to be injected into metacard types. For example, attributes defined in JSON definition files are placed in the registry, so they can be injected.
```
Comment (mcalcote) [inline distribution/docs/src/main/resources/content/_devComponents/attribute-injection.adoc]: ✏️ Nit: `... interface) to be injected ...`

**Comment by ricklarsen** [comment]
build connection issues. rebased & rerunning build now

**Comment by LinkMJB** [comment]
Those errors look related to https://github.com/netty/netty/issues/4671 -- I don't think it was a connectivity issue.

**Comment by ricklarsen** [comment]
@LinkMJB That makes more sense. I had never seen that particular error before

**Comment by ricklarsen** [comment]
build now

**Pull request merged**